File: com.unity.uiwidgets/Runtime/ui/renderer/compositeCanvas/flow/backdrop_filter_layer.cs
using Unity.UIWidgets.foundation;
using Unity.UIWidgets.ui;

namespace Unity.UIWidgets.flow {
    public class BackdropFilterLayer : ContainerLayer {
        ImageFilter _filter;

        public ImageFilter filter {
            set { _filter = value; }
        }

        public override void paint(PaintContext context) {
            D.assert(needsPainting);

            var canvas = context.canvas;
            canvas.saveLayer(paintBounds, new Paint {backdrop = _filter});

            try {
                paintChildren(context);
            }
            finally {
                canvas.restore();
            }
        }
    }
}

File: engine/src/flow/layers/backdrop_filter_layer.cc
#include "flow/layers/backdrop_filter_layer.h"

namespace uiwidgets {

BackdropFilterLayer::BackdropFilterLayer(sk_sp<SkImageFilter> filter)
    : filter_(std::move(filter)) {}

void BackdropFilterLayer::Preroll(PrerollContext* context,
                                  const SkMatrix& matrix) {
  Layer::AutoPrerollSaveLayerState save =
      Layer::AutoPrerollSaveLayerState::Create(context, true, bool(filter_));
  ContainerLayer::Preroll(context, matrix);
}

void BackdropFilterLayer::Paint(PaintContext& context) const {
  TRACE_EVENT0("uiwidgets", "BackdropFilterLayer::Paint");
  FML_DCHECK(needs_painting());

  Layer::AutoSaveLayer save = Layer::AutoSaveLayer::Create(
      context,
      SkCanvas::SaveLayerRec{&paint_bounds(), nullptr, filter_.get(), 0});
  PaintChildren(context);
}

}  // namespace uiwidgets
